sklearn/ensemble/_gb.py

```diff
@@ -578,7 +578,16 @@ def fit(self, X, y, sample_weight=None, monitor=None):
             # The requirements of _decision_function (called in two lines
             # below) are more constrained than fit. It accepts only CSR
             # matrices.
-            X = check_array(X, dtype=DTYPE, order="C", accept_sparse="csr")
+            if self.warm_start:
+                X = check_array(
+                    X,
+                    dtype=DTYPE,
+                    order="C",
+                    accept_sparse="csr",
+                    force_all_finite=False,
+                )
```

Comment:
I removed the condition on this `check_array` and applied your changes to L732.

I couldn't completely remove L582-590 because sparse matrix needs to be csr. e4b58e0a1 is related to this